arangod/Cluster/Maintenance.cpp

```diff
@@ -1410,6 +1529,86 @@ arangodb::Result arangodb::maintenance::reportInCurrent(
       throw;
     }
 
+    // UpdateReplicatedLogs
+    try {
+
+      if (auto logsIter = localLogs.find(dbName); logsIter != std::end(localLogs)) {
+        for (auto const& [id, status] : logsIter->second) {
+          // Check Current/ReplicatedLogs/<dbname>/<logId>/localStatus/<serverId>/currentTerm != currentTerm
+          // if so, update Current/ReplicatedLogs/<dbname>/<logId>/localStatus/<serverId> with
+          //  {"currentTerm": currentTerm, "spearHead": {"index": last-index, "term": last-term}}
+          // and precondition
+          //  Plan/ReplicatedLogs/<dbname>/<logId>/term/term == currentTerm
+          auto localTerm = status.getCurrentTerm();
+          if (!localTerm.has_value()) {
+            continue;
+          }
+
+          auto termInCurrent = cur.get(cluster::paths::aliases::current()
+                                           ->replicatedLogs()
+                                           ->database(dbName)
+                                           ->log(to_string(id))
+                                           ->localStatus()
+                                           ->participant(serverId)
+                                           ->term()
+                                           ->vec());
+
+          if (!termInCurrent.isNone() &&
+              replication2::LogTerm{termInCurrent.extract<uint64_t>()} ==
+                  localTerm.value()) {
+            // nothing to do
+            continue;
+          }
+
+          auto logContext =
+              LoggerContext{Logger::MAINTENANCE}.with<logContextKeyLogId>(id);
+
+          LOG_CTX("11dbd", TRACE, logContext)
+              << "checking replicated log " << id << " local term = "
+              << (localTerm ? to_string(localTerm.value()) : "n/a") << " current "
```

Comment:
`localTerm` will always have a value here. This is guaranteed by the check in line 1543.